geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -318,5 +320,27 @@ object FilterHelper {
       case f: Filter => Seq(f)
     }
   }
+
+  def tryMergeGeometryFilters(filts: Seq[Filter]): Seq[Filter] = {
+    import org.geotools.data.DataUtilities._
+    import scala.collection.JavaConversions._
+    val filtFactory = CommonFactoryFinder.getFilterFactory2
+
+    def getAttrName(l: BBOX): String = propertyNames(l).head.getPropertyName
+
+    filts match {
+      // if we have two bbox filters as is common in WMS queries, merge them by intersecting the bounds
+      case Seq(l: BBOX, r: BBOX) if getAttrName(l) == getAttrName(r) =>
```

Comment:
It feels as if it might be useful to extend this to work with longer sequences, even if our initial case is known to have only two elements.  Thoughts?